### PR TITLE
Adding the nodejs0.12 auth option

### PR DESCRIPTION
**Pull request by ashubham**
On the basis of https://nodejs.org/api/https.html#https_https_request_options_callback

### CODE REVIEW

File: lib/http-proxy/common.js [hunk 1]
```diff
@@ -46,6 +46,10 @@ common.setupOutgoing = function(outgoing, options, req, forward) {
     extend(outgoing.headers, options.headers);
   }
 
+  if(options.auth){
```
Comment (3rd-Eden) [inline lib/http-proxy/common.js]: Please follow the code style in this document (adding spaces around the parentheses)

File: test/lib-http-proxy-common-test.js [hunk 2]
```diff
@@ -17,6 +17,7 @@ describe('lib/http-proxy/common.js', function () {
         },
         headers: {'fizz': 'bang', 'overwritten':true},
         localAddress: 'local.address',
+	auth:'username:pass'
```
Comment (3rd-Eden) [inline test/lib-http-proxy-common-test.js]: Use spaces instead of tabs for indenting here.

**Comment by 3rd-Eden** [comment]
It could use a test to ensure that the Authorization header is actually set instead of checking if the options are supplied.

**Comment by jcrugzz** [comment]
LGTM

**Pull request merged**